### DISCUSSION TITLE
Homepage integration (access web api)

### DISCUSSION THREAD

**Discussion by schmitzCatz** [PlexRipper/PlexRipper#350] [Ideas]
Hi,

I was wondering if it is possible to get plexripper stats for a [Hompage widget](https://gethomepage.dev/widgets/) via [Custom API Wiget](https://gethomepage.dev/widgets/services/customapi/). I peeked a bit into documentation and source code and realized frontend is written in VUE and backend in C#. I'm unfamiliar with the .net framework and was unable to find any documentation about the API the frontend is using. Is there an openAPI doc or something?

To make a long story short, I there any cheap way to access the api? Would be nice to see some stats at the dashboard like running downloads, number of server etc. :-)

Thanks and keep up the awesome work!

**Comment by JasonLandbridge** [comment]
Hi there!

Currently there is no external API, but this will come eventually. I need to create an external API for Radarr/Sonarr integration anyway, which could then be used for a widget as well. And the internal API will be locked down to prevent hacks/abuse, so for now that is not really an option.

What other kind of specific info would you like to see? PlexRipper has been recently integrated in [Heimdall](https://github.com/linuxserver/Heimdall-Apps/tree/master/PlexRipper), any other apps I can help out with?

> Thanks and keep up the awesome work!

Appreciate it! Cheers!

**Comment by schmitzCatz** [comment]
Hi,

>> What other kind of specific info would you like to see?

Good question, from the top of my hat:

* Number of Libraries [Show, Movie, Audio, Pictures]
* Number of Media per Library
* Number of Servers
* Number of Accounts
* Number of Downloads
* Number of Notifications

With access to a public api, it is possible to easily integrate Ripper in Homepage. 

I use [Gaps](https://github.com/JasonHHouse/Gaps) long term I would be nice to see such functionality within PlexRipper, since Gaps is not in active development anymore :-(


Cheers!
